frontend/src/pages/Rankings.tsx

```diff
@@ -0,0 +1,134 @@
+import {useTranslation} from "react-i18next";
+import {
+    Box,
+    Container, Typography,
+} from "@mui/material";
+import * as React from 'react';
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+import Stack from '@mui/material/Stack';
+import Switch from '@mui/material/Switch';
+import CircularProgress from "@mui/material/CircularProgress";
+
+import {eventID, Province, provinceID, useAverage} from "../components/Types";
+import {GetProvinces} from "../components/Provinces";
+import {useEffect, useState} from "react";
+import {API_BASE_URL} from "../components/Api";
+import httpClient from "../httpClient";
+import {RankList} from "../components/RankList";
+
+
+export const Rankings = () => {
+    const {t} = useTranslation();
+
+    const provinces: Province[] = GetProvinces();
```

Comment:
If a constant doesn't depend on anything in the component, you can lift it out (to line 20). Then it's the same value every render, instead of being recalculated. This saves some computation (minimal in this case) but also ensures dependency arrays don't think it's a new value every render.